Add a WIP banner to product list

anitaa1990 opened the pull request:
Fixes #1602 . This PR adds a new work in progress banner to the product list screen.

### Screenshots
<img width="300" src="https://user-images.githubusercontent.com/22608780/69403328-4a024a00-0d20-11ea-9cf1-cf60dc0e64ab.png"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/69403334-4bcc0d80-0d20-11ea-917d-849bcd64000f.png">

#### Empty product state
<img width="300" src="https://user-images.githubusercontent.com/22608780/69403821-9601be80-0d21-11ea-8bb2-3315fb629720.png"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/69403826-97cb8200-0d21-11ea-857f-37ab7dd69225.png">

### Testing:
- Verify that the WIP banner is only displayed when the product is list loaded (and not when skeleton view is displayed).
- Verify that the banner is displayed even when there are no products in the store.
- Verify that the banner is displayed correctly in lollipop devices.


Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

anitaa1990 commented:
Thanks for the review @kyleaparker!

> The progress banner needs to be separated from the card below so that there is some definition between the two (refer to the designs)

Updated in the app. Below are the latest screenshots:

<img width="300" src="https://user-images.githubusercontent.com/22608780/69606630-513f9580-1049-11ea-98d8-97b1fb05e285.png"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/69606636-53a1ef80-1049-11ea-8f14-28c6fdfec19c.png">

#### Empty product state
<img width="300" src="https://user-images.githubusercontent.com/22608780/69606652-6288a200-1049-11ea-8e22-5bfe0ba881ad.png"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/69606656-64eafc00-1049-11ea-9b0d-a79c90ffc439.png">

nbradbury commented:
@anitaa1990 Code looks good! Let me know when this passes design review and I'll :shipit:

anitaa1990 commented:
> I'm seeing a faint border extending to the left side on each row that should be removed:

Nice catch! Fixed and updated with below screenshot 👍 
<img width="300" src="https://user-images.githubusercontent.com/22608780/69649571-3c87ef80-1093-11ea-9210-5fab4c82e28c.png">
